lib/util/fs.js

```diff
@@ -115,6 +115,11 @@ exports.dirname = dirname;
  * @returns {void}
  */
 const mkdirp = (fs, p, callback) => {
+	if (path.resolve(p) === path.resolve("/")) {
```

Comment:
Do you really need so many `path.resolve`? They have a performance cost. I think `p` is always a absolute path anyway. What is about windows absolute paths?

```suggestion
	if (/^(\/|[A-Za-z]:\\?|\\\\[^\\]+\\?)$/.test(p)) {
```

Reply:
This will fail edge cases like multiple trailing backslashes (such as `C:\\\\`).

Though, we can avoid the performance overhead by moving the logic into the error handling part.